api/envoy/api/v2/route/route.proto

```diff
@@ -846,6 +859,24 @@ message RetryPolicy {
   repeated uint32 retriable_status_codes = 7;
 }
 
+// HTTP request hedging TODO(mpuncel) docs
+// [#not-implemented-hide:]
+message HedgePolicy {
+  // Specifies the number of initial requests that should be sent upstream.
+  // Must be at least 1.
+  // Defaults to 1.
+  google.protobuf.UInt32Value initial_requests = 1 [(validate.rules).uint32.gte = 1];
+
+  // Specifies a probability that an additional upstream request should be sent
+  // on top of what is specified by initial_requests. Defaults to 0.
+  envoy.type.Percent additional_request_chance = 2;
```

Comment:
working on the next step of this, i'm wondering if this is the right type to use here.

Traffic shadowing uses `core.RuntimeFractionalPercent`, which seems to have more helper methods and also would allow changing the value via runtime. However it'd be kinda weird if you can only change the percentage and not initial_requests.

Sampling percentages in HTTP connection manager use `core.type.Percent` like this, and weighted clusters use a uint for weighting each cluster and also a `total_weight` for the denominator. Then there's also `envoy.type.FractionalPercent` which looks like also has more helpers

Reply:
I would probably use `envoy.type.FractionalPercent` and add more runtime/random helpers if you need to. I think it would also be perfectly fine to use `RuntimeFractionalPercent` if you also want to allow runtime control. Up to you.

Reply:
seems like that will work out nicely